yarn-project/prover-client/src/job_pool/broker_backend/in_memory.ts

```diff
@@ -0,0 +1,59 @@
+import assert from 'assert/strict';
+
+import type { ProofOutputs, ProofRequest, ProofRequestId, ProofResult, ProofType } from '../proof_request.js';
+import { type BrokerBackend } from './interface.js';
+
+export class InMemoryBrokerBackend implements BrokerBackend {
+  private items = new Map<ProofRequestId, ProofRequest<ProofType>>();
+  private results = new Map<ProofRequestId, ProofResult<ProofType>>();
```

Comment:
In my mind, we had two very different backends: one for storing metadata (proof id, type, status, etc) which was used for recovering from a crash, and one for storing payloads (inputs and results) which was shared with the agents. Is the plan to split things up later?

Reply:
Oh I see, right now everything goes through the proving broker but we could also bypass it and only pass URIs to proofs. I'll refactor this in a separate PR (gonna open a github issue for it too)